### PR TITLE
Query : Get Modules List

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Added support for query `GetModuleList`

## Changes
Added support for query `GetModuleList`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

Depends on #89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -373,6 +379,12 @@ val peers = client.getPeerList(true);
 SemVer version = client.getVersion();
 ```
 
+#### getModuleList
+```java
+val list = client
```
Comment (EmilLa1) [inline README.md:384]: Please specify type.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 2]
```diff
@@ -313,6 +314,25 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the list of smart contract modules in the given block.
+     * @param blockHash {@link Hash} of block at which the modules list is being retrieved.
+     * @return Parsed {@link ImmutableList} of {@link Hash}
+     * @throws Exception When the returned JSON is null.
+     */
+    public ImmutableList<Hash> getModuleList(final Hash blockHash) throws Exception {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:323]: I think you want to return `ImmutableList<ModuleRef>`?

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/modulelist/ModuleRef.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.concordium.sdk.serializing.JsonMapper;
+import com.concordium.sdk.transactions.Hash;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ModuleRef {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/modulelist/ModuleRef.java:12]: I think you want to wrap a `Hash` ?
I.e. a module reference is simply a sha256 hash.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Please add unit test (for parsing json).

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/modulelist/ModuleRef.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/modulelist/ModuleRef.java:12]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 2 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:323]: done

File: README.md (hunk 1 above)
Comment (parv0888) [reply README.md:384]: done

**Comment by parv0888** [comment]
> Please add unit test (for parsing json).

done

**Comment by EmilLa1** [comment]
Closed via https://github.com/Concordium/concordium-java-sdk/pull/91

**Pull request closed without merging**